.github/workflows/alarm.yml

```diff
@@ -18,9 +21,364 @@ jobs:
+          MAX_REVISIONS = int(os.environ.get("MAX_REVISIONS","5"))
+          NO_CHANGE_PATIENCE = int(os.environ.get("NO_CHANGE_PATIENCE","100"))
+
+          # ----- load artifact (current results) -----
+          artifacts_path = os.path.abspath(os.environ.get("ARTIFACTS_DIR", "./artifacts"))
+          if not os.path.exists(artifacts_path):
+              open(os.environ.get("PR_COMMENT_PATH","pr_comment.md"),"w").close()
+              open(os.environ.get("CHECK_BODY_PATH","check_output.md"),"w").close()
+              sys.exit(0)
+
+          current_txt_path = None
+          for root, _, files in os.walk(artifacts_path):
+              for fname in files:
+                  if fname.startswith("speed_test") and fname.endswith(".txt"):
+                      current_txt_path = os.path.join(root, fname); break
+              if current_txt_path: break
+          if current_txt_path is None:
+              open(os.environ.get("PR_COMMENT_PATH","pr_comment.md"),"w").close()
+              open(os.environ.get("CHECK_BODY_PATH","check_output.md"),"w").close()
+              sys.exit(0)
+
+          with open(current_txt_path, "r", encoding="utf-8") as f:
+              current_bm = artifacts_parse_speed_txt_lines(f.readlines())
+
+          # ----- W&B baselines -----
+          if not os.getenv("WANDB_API_KEY"):
+              try: wandb.login(anonymous="allow", relogin=True)
+              except Exception: pass
+          ENTITY = os.environ.get("WANDB_ENTITY","")
+          PROJECT= os.environ.get("WANDB_PROJECT","")
+
+          api = wandb.Api()
+          runs_iter = api.runs(f"{ENTITY}/{PROJECT}", order="-created_at")
+
+          by_rev = {}; rev_order = []; selected_revs=None; no_change=0
+          for run in runs_iter:
+              if run.state != "finished": continue
+              cfg = getattr(run, "config", None)
+              if cfg is None: continue
+              cfg = json.loads(cfg)
+              raw_rev = cfg.get("revision"); raw_bid = cfg.get("benchmark_id")
+              if not raw_rev or not raw_bid:
+                  if selected_revs is not None:
+                      no_change += 1
+                      if no_change >= NO_CHANGE_PATIENCE: break
+                  continue
+              rev = _norm_rev(raw_rev.get("value")); bid = raw_bid.get("value")
+              if not rev or not bid:
+                  if selected_revs is not None:
+                      no_change += 1
+                      if no_change >= NO_CHANGE_PATIENCE: break
+                  continue
+              if selected_revs is not None and rev not in selected_revs:
+                  no_change += 1
+                  if no_change >= NO_CHANGE_PATIENCE: break
+                  continue
+              if rev not in by_rev:
+                  by_rev[rev]={}; rev_order.append(rev)
+                  if len(rev_order) >= MAX_REVISIONS: selected_revs = set(rev_order)
```

Comment:
I don't understand why this is so complicated. On my side, I can just read the summary. Which information are you missing?
```python
records = []
for run in tqdm(runs, desc="Fetching run summaries"):
    if not run.state == "finished":
        continue
    config = run.config
    summary = run.summary

    records.append(
        {
            "benchmark_id": config["benchmark_id"],
            "timestamp": pd.to_datetime(config["timestamp"], unit="s"),
            "revision": config["revision"],
            "runtime_fps": summary.get("runtime_fps", float("nan")),
            "compile_time": summary.get("compile_time", float("nan")),
        }
    )

```

Reply:
Ok let me have a look.